src/plugins/intel_gpu/src/plugin/device_config.cpp

```diff
@@ -556,6 +556,23 @@ std::string Config::ConvertPropertyToLegacy(const std::string& key, const std::s
     IE_THROW() << "Unsupported value for legacy key : " << key;
 }
 
+bool Config::operator==(const Config& other) const {
```

Comment:
I think such operator== is a bit confusing as it actually checks options that impacts context reuse only, not all of them